### PR TITLE
[Dashboard] Abstract code for CSV upload form

### PR DESCRIPTION
**Pull request by kien-ngo**
DASH-372

---

## PR-Codex overview
This PR focuses on refactoring the `AirdropUpload` and `SnapshotUpload` components to utilize a new `useCsvUpload` hook for handling CSV uploads and data processing. It also introduces a new `CsvDataTable` component for displaying data in a tabular format.

### Detailed summary
- Added `useCsvUpload` hook for handling CSV uploads.
- Introduced `CsvDataTable` for rendering tabular data.
- Refactored `AirdropUpload` to use `useCsvUpload` and `CsvDataTable`.
- Simplified CSV parsing and validation logic.
- Removed redundant code related to address validation.
- Updated `SnapshotUpload` similarly to `AirdropUpload`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
* **#5324** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5324?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/tokens/components/airdrop-upload.tsx [hunk 1]
```diff
@@ -1,37 +1,15 @@
...
-import { type DropzoneOptions, useDropzone } from "react-dropzone";
-import { type Column, usePagination, useTable } from "react-table";
-import { type ThirdwebClient, ZERO_ADDRESS } from "thirdweb";
-import { resolveAddress } from "thirdweb/extensions/ens";
-import { isAddress } from "thirdweb/utils";
+import { Link } from "@chakra-ui/react";
+import { useCsvUpload } from "hooks/useCsvUpload";
+import { CircleAlertIcon, UploadIcon } from "lucide-react";
+import { useMemo, useRef } from "react";
+import {} from "react-dropzone";
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/tokens/components/airdrop-upload.tsx:8]: empty import

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/_components/claim-conditions/snapshot-upload.tsx [hunk 2]
```diff
@@ -1,43 +1,23 @@
...
-import Papa from "papaparse";
-import { useCallback, useEffect, useRef, useState } from "react";
-import { type DropzoneOptions, useDropzone } from "react-dropzone";
-import { type Column, usePagination, useTable } from "react-table";
-import { isAddress } from "thirdweb";
-import { resolveAddress } from "thirdweb/extensions/ens";
+import { useCsvUpload } from "hooks/useCsvUpload";
+import { CircleAlertIcon, DownloadIcon, UploadIcon } from "lucide-react";
+import { useCallback, useRef } from "react";
+import {} from "react-dropzone";
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/_components/claim-conditions/snapshot-upload.tsx:16]: empty import

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/_components/csv-data-table.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    canPreviousPage,
+    canNextPage,
+    pageOptions,
+    pageCount,
+    gotoPage,
+    nextPage,
+    previousPage,
+    setPageSize,
+    state: { pageIndex, pageSize },
+  } = useTable(
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/_components/csv-data-table.tsx:52]: if we're updating this component lets please update this to [tanstack table](https://tanstack.com/table/latest/docs/introduction) while we're at it

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/tokens/components/airdrop-upload.tsx (hunk 1 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/tokens/components/airdrop-upload.tsx:8]: oh..

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/_components/csv-data-table.tsx (hunk 3 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/_components/csv-data-table.tsx:52]: Sure. I'll stack up the PRs

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/_components/claim-conditions/snapshot-upload.tsx (hunk 2 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/_components/claim-conditions/snapshot-upload.tsx]: removed

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/tokens/components/airdrop-upload.tsx (hunk 1 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/tokens/components/airdrop-upload.tsx]: removed!

**Review by jnsdls — APPROVED** [review]

**Pull request merged**